src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -389,52 +389,56 @@ private static void CheckSnapshot(ITextSnapshot snapshot)
 
         private ImmutableDictionary<ITextBuffer, TagSpanIntervalTree<TTag>> ComputeNewTagTrees(ImmutableDictionary<ITextBuffer, TagSpanIntervalTree<TTag>> oldTagTrees, TaggerContext<TTag> context)
         {
-            // Ignore any tag spans reported for any buffers we weren't interested in.
-
-            var spansToTag = context.SpansToTag;
-            var buffersToTag = spansToTag.Select(dss => dss.SnapshotSpan.Snapshot.TextBuffer).ToSet();
-            var newTagsByBuffer =
-                context.TagSpans.Where(ts => buffersToTag.Contains(ts.Span.Snapshot.TextBuffer))
-                                .ToLookup(t => t.Span.Snapshot.TextBuffer);
```

Comment:
lots of wasteful linq/enumerator code here.  it all got changes to simple code using pools.